src/import/sst_service.rs

```diff
@@ -548,12 +573,76 @@ where
                 })
             }
             .await;
-            send_rpc_response!(res, sink, label, timer);
+            crate::send_rpc_response!(res, sink, label, timer);
         };
 
         self.threads.spawn_ok(buf_driver);
         self.threads.spawn_ok(handle_task);
     }
+
+    fn duplicate_detect(
+        &mut self,
+        _ctx: RpcContext<'_>,
+        mut request: DuplicateDetectRequest,
+        mut sink: ServerStreamingSink<DuplicateDetectResponse>,
+    ) {
+        let label = "duplicate_detect";
+        let timer = Instant::now_coarse();
+        let context = request.take_context();
+        let router = self.router.clone();
+        let start_key = request.take_start_key();
+        let end_key = if request.get_end_key().is_empty() {
+            None
+        } else {
+            Some(request.take_end_key())
+        };
+        let key_only = request.get_key_only();
+        let snap_res = Self::async_snapshot(router, make_request_header(context));
+        let handle_task = async move {
+            let res = snap_res.await;
+            let snapshot = match res {
+                Ok(snap) => snap.snapshot,
+                Err(e) => {
+                    let mut resp = DuplicateDetectResponse::default();
+                    pb_error_inc(label, &e);
+                    resp.set_region_error(e);
+                    match sink
+                        .send((resp, WriteFlags::default().buffer_hint(true)))
+                        .await
+                    {
+                        Ok(_) => {
+                            IMPORT_RPC_DURATION
+                                .with_label_values(&[label, "ok"])
+                                .observe(timer.elapsed_secs());
+                        }
+                        Err(e) => {
+                            warn!(
+                                "connection send message fail";
+                                "err" => %e
+                            );
+                        }
+                    }
+                    let _ = sink.close().await;
+                    return;
+                }
+            };
+            let detector = DuplicateDetector::new(snapshot, start_key, end_key, key_only).unwrap();
```

Comment:
It seem `min_commit_ts` in request is not used?

Reply:
Yes.... I just left it for incremental-import